app/gui/view/graph-editor/src/component/edge.rs

```diff
@@ -4,2033 +4,1044 @@ use crate::prelude::*;
-            ensogl::shape! {
-                below = [joint];
-                alignment = center;
-                (
-                    style: Style,
-                    radius: f32,
-                    angle: f32,
-                    start_angle: f32,
-                    pos: Vector2<f32>,
-                    dim: Vector2<f32>,
-                    focus_split_center: Vector2<f32>,
-                    focus_split_angle: f32,
-                    color_rgba: Vector4<f32>,
-                    focus_color_rgba: Vector4<f32>,
-                ) {
-                    let width       = &LINE_WIDTH.px();
-                    let shape       = corner_base_shape(&radius,width,&angle,&start_angle);
-                    let color       = Var::<color::Rgba>::from(color_rgba);
-                    let focus_color = Var::<color::Rgba>::from(focus_color_rgba);
-
-                    let shadow_size = 10.px() + 1.px();
-                    let node_radius = &shadow_size + 1.px() * dim.y();
-                    let node_shape  = Rect((&shadow_size*2.0 + 2.px() * dim.x(),&node_radius*2.0));
-                    let node_shape  = node_shape.corners_radius(node_radius);
-                    let node_shape  = node_shape.fill(color::Rgba::new(1.0,0.0,0.0,1.0));
-                    let tx       = - 1.px() * pos.x();
-                    let ty       = - 1.px() * pos.y();
-                    let node_shape  = node_shape.translate((tx,ty));
-
-                    let shape = shape.intersection(node_shape);
-
-                    let split_shape = FocusedEdge::new(
-                    shape,&focus_split_center.px(),&focus_split_angle.into());
-                    let shape       = split_shape.fill(&color,&focus_color);
-
-                    let hover_width = width + HOVER_EXTENSION.px() * 2.0;
-                    let hover_area  = corner_base_shape(&radius,&hover_width,&angle,&start_angle);
-                    let hover_area  = hover_area.fill(HOVER_COLOR);
-                    (hover_area + shape).into()
-                }
-            }
+    fn set_mouse_position_and_redraw(&self, screen_pos: Vector2<f32>) {
+        // Convert point to local coordinates.
+        let screen_pos_3d = Vector3(screen_pos.x(), screen_pos.y(), 0.0);
+        let scene_pos = self.scene.screen_to_scene_coordinates(screen_pos_3d).xy();
+        let pos = scene_pos - self.display_object.xy();
 
-            impl EdgeShape for View {
-                fn set_focus_split_center_local(&self, center: Vector2<f32>) {
-                    self.focus_split_center.set(center);
-                }
+        self.hover_position.set(Some(pos));
+        self.redraw();
+    }
 
-                fn set_focus_split_angle(&self, angle: f32) {
-                    self.focus_split_angle.set(angle);
-                }
+    fn new_section(&self) -> Rectangle {
+        let new = Rectangle::new();
+        new.set_corner_radius_max();
+        new.set_inset_border(LINE_WIDTH);
```

Comment:
These functions should have at least small docs. Based on this name, it's hard to understand what a "section" is.